ui/src/screens/Post/components/NewPostForm.tsx

```diff
@@ -65,12 +76,48 @@ const NewPostForm: React.FC = function () {
         toast("We are unable to create your post at this time 🙇‍♂️");
       }
     };
-    postJob();
+
+  const updateJob = async () => {
+    try {
+      const response = await JobsGoWhereApiClient.put(
+        `${process.env.REACT_APP_API}/${values.type}sbyid/${postContext.post?.id}`,
+        JSON.stringify(values),
+        {
+          headers: {
+            "Content-Type": "application/json",
+          },
+        },
+      )
+
+      toast("Your post has been successfully updated! 👍");
+      await new Promise((response) => setTimeout(response, 3000));
+      postContext.setPost(null)
+      postContext.setType(null)
+      history.push(`/${values.type}s`);
+    } catch (err) {
+      console.error("error", err);
+      toast("We are unable to update your post at this time 🙇‍♂️");
+    }
+  }
+
+    if(postContext.post?.id && location.pathname === EDIT_POST_PATHNAME) {
+      updateJob();
+    } else {
+      postJob();
+    }
   };
 
   React.useEffect(() => {
     register({ name: "type" }, { required: true });
-    setValue("type", INITIAL_TYPE);
+    if(postContext.type) {
+      setValue("type", postContext.type.slice(0,-1))
```

Comment:
whats the situation where this branch triggers?

Reply:
In the event you are editting your post, this will set the radio buttons to the correct post type.